app/packs/components/workflows/student/exams/show/components/VerticalScrollShadow.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useMemo, useRef, useState } from 'react';
+import './VerticalScrollShadow.scss';
+
+const VerticalScrollShadow: React.FC<{
+  className?: string,
+}> = (props) => {
+  const {
+    className,
+    children,
+  } = props;
+  const parentRef = useRef<HTMLDivElement>(null);
+  const scrollRef = useRef<HTMLDivElement>(null);
+  const [atTop, setAtTop] = useState(true);
+  const [atBot, setAtBot] = useState(true);
+  const marginMarkers = { topMarker: setAtTop, botMarker: setAtBot };
+  const observer = useMemo(() => (
+    new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        marginMarkers[entry.target.className](entry.isIntersecting);
+      });
+    }, {
+      root: parentRef.current,
+      rootMargin: '0px',
+      threshold: 1,
+    })
+  ), []);
+  useMemo(() => {
```

Comment:
this `useMemo`'s result is not used? Was this supposed to be `useEffect` -- does this code get called at all right now?

Reply:
Actually...I'm a bit more skeptical that this works correctly all the time.  It sure seems to work when I ran it, but -- when the memoized `observer` gets created, `parentRef.current == null`, right?

I'm not sure whether I want to debug this myself, or to use https://github.com/thebuilder/react-intersection-observer instead ...

Reply:
Yeah, that's part of why I'm concerned that these few lines of code are all slightly wrong.  Once that changes the `observer`, that prevents me from ever disconnecting the previous observer, leading to resource leaks.  `react-intersection-observer` seems to handle this, in that its `useInView` returns an `unobserve` callback, but I haven't found any examples of using that library with an explicit root and target element (i.e. _two_ node `ref`s), so I'm not 100% sure how to use that library yet...

Reply:
The function you pass to `useEffect` may return another callback which does unloading cleanup -- so you can disconnect the observer when it changes in that callback before the next one is hooked up

Reply:
I don't know anything about observers.. but for setInterval this might look like:


```javascript
useEffect(() => {
  let interval = setInterval(...);
  return () => {
    clearInterval(interval);
  };
}, [someDependency]);
```
In the example above, if `someDependency` changes, the old interval is cleaned up and a new one is created.
Does this help at all?

Reply:
Yeah; see latest commit, I think that fixes things properly...